### DISCUSSION TITLE
Sending / Receiving commands from tasmota to my linux PC

### DISCUSSION THREAD

**Discussion by stulluk** [arendst/Tasmota#14602] [Q&A] [answered]
I am using tasmota on a NEO Coolcam WIFI plug.

Sadly, it is extremely hard to install tasmota to this device. See https://www.youtube.com/watch?v=CV8Y61gEQpQ  I killed two devices while doing this.

1 year ago, I used tuya-convert to be able to install tasmota to this device, and using since then, very reliably. But since last 6 months, tuya-convert doesn't work anymore on these devices sadly. See this: https://github.com/ct-Open-Source/tuya-convert/issues/1022

One of these devices control my Desktop PC running linux. This PC has SSH server.

My question is, would it be possible to write a script on Tasmota WEB UI, so that before cutting OFF the power, it can send gracefull shutdown command ( Such as "sudo poweroff" ) over SSH ? If SSH is not possible, even raw strings ? ( I can listen from specific port via netcat or write a simple C app for it) 

Is this possible?

**Comment by joba-1** [comment]
> My question is, would it be possible to write a script on Tasmota WEB UI, so that before cutting OFF the power, it can send gracefull shutdown command ( Such as "sudo poweroff" ) over SSH ? If SSH is not possible, even raw strings ? ( I can listen from specific port via netcat or write a simple C app for it)
> 
> Is this possible?

Ssh is not possible, but tasmota can send http requests (websend). You could interpret that as "raw strings" in your linux app. Or you could run a webserver - even on a specific nonstandard port - receiving and acting on that.
Tasmota can also send mqtt messages (publish) and your linux pc could subscribe to them.

I guess you would need to use a virtual switch triggering rules to monitor power and switch off once no power is consumed (assuming the device can monitor power like my Nous that looks the same). Haven't used this, so thats where my expertise ends...

**Comment by sfromis** [comment, marked as answer]
Using the command WebQuery (less complex variant of WebSend), you can connect to some http URL.
